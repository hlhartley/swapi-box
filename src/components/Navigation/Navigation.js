import React from 'react';
import '../../main.scss';

const Navigation = (props) => {
    return (
        <div className={ props.navFixed ? "navigation-container fixed-nav" : "navigation-container" }>
            <div className='logo'>SWAPI-BOX</div>
            <a className='people-link' href='www.google.com'><i className="fas fa-male"></i> PEOPLE [{props.people.length}]</a>
            <a className='planets-link' href='www.google.com'><i className="fas fa-globe"></i> PLANETS</a>
            <a className='vehicles-link' href='www.google.com'><i className="fas fa-fighter-jet"></i> VEHICLES</a>
            <a className='favorites-link' href='www.google.com'><i className="far fa-star"></i> FAVORITES [0]</a>
        </div>
    )
}

export default Navigation;