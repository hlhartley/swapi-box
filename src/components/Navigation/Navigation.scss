.navigation-container {
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: #011c44;
    color: #e2dbd3;
    height: 87px;
    position: relative;
    top: 0;
    transition: all 0.5s;
    z-index: 1;
}

a {
    color: #e2dbd3;
    text-decoration: none;
    letter-spacing: 1px;
}

a:visited {
    color: #e2dbd3;
}

a:hover {
    color: #8895ab;
}

.fixed-nav {
    display: flex;
    align-items: center;
    justify-content: space-around;
    position: fixed;
    box-shadow: 0 5px rgba(0, 0, 0, 0.1);
    background-color: black;
    width: 100%;
}

.logo {
    max-width: 0;
    overflow: hidden;
    background: white;
    transition: all 0.4s;
    font-weight: 600;
    color: black;
}

.fixed-nav .logo {
    display: flex;
    flex-direction: column; 
    max-width: 500px;
    font-size: 22px;
    justify-content: center;
    align-items: center;
    margin: 10px;
    width: 150px;
    height: 50px;
}

.people-link,
.planets-link,
.vehicles-link,
.favorites-link {
    display: flex;
    flex-flow: column;
    justify-content: center;
    align-items: center;
    line-height: 35px;
    // border-left: 1px solid #e2dbd3;
    // border-right: 1px solid #e2dbd3;
    padding: 10px;
    margin-left: 20px;
    margin-right: 20px;
    width: 200px;
}

.fa-male {
    color: #36898d;
}

.fa-globe {
    color: #9e4e5f;
}

.fa-fighter-jet {
    color: #419959;
}

.fa-star {
    color: #c6c549;
}

.fas,
.far {
    font-size: 25px;
}