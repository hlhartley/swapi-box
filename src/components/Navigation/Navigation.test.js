import React from 'react';
import ReactDOM from 'react-dom';
import Navigation from './Navigation';
import { shallow } from 'enzyme';

let wrapper;
describe('Navigation', () => {

  beforeEach(() => {
    wrapper = shallow(<Navigation/>); 
  })

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Navigation />, div);
    ReactDOM.unmountComponentAtNode(div);
})


  it('should exist', () => {   
    expect(wrapper).toBeDefined()
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  // it.skip('should have a proper default state', () => {
  //   expect(wrapper.state()).toEqual({
      
  //   })
  })
