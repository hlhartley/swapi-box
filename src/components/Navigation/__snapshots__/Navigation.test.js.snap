// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should match the snapshot with all data passed in correctly 1`] = `
<div
  className="navigation"
>
  Navigation
</div>
`;
